From df135afaac88d6ea5131d09136c0734ff56d00fb Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sat, 15 Apr 2023 16:20:59 +0200
Subject: [PATCH 1/2] Add several new functions from treble settings

Add netflix bsp_rev
Add several new functions
- Add safetynet/securize
- Add spoof model with custom properties
- Add Autorun
- Add NoLog
- Add Restart sysui
- Add Dump Logs
---
 base.mk             |  1 +
 phh-on-data.sh      | 17 +++++++---
 phh-prop-handler.sh | 80 +++++++++++++++++++++++++++++++++++++++++++++
 rw-system.sh        | 61 +++++++++++++++++++++++++++++++++-
 secure.sh           | 16 +++++++++
 vndk.rc             | 19 +++++++++++
 6 files changed, 189 insertions(+), 5 deletions(-)
 create mode 100644 secure.sh

diff --git a/base.mk b/base.mk
index ed07565..ef61dcc 100644
--- a/base.mk
+++ b/base.mk
@@ -87,6 +87,7 @@ PRODUCT_COPY_FILES += \
 	device/phh/treble/rw-system.sh:system/bin/rw-system.sh \
 	device/phh/treble/phh-on-data.sh:system/bin/phh-on-data.sh \
 	device/phh/treble/phh-prop-handler.sh:system/bin/phh-prop-handler.sh \
+	device/phh/treble/secure.sh:system/phh/secure.sh \
 	device/phh/treble/fixSPL/getSPL.arm:system/bin/getSPL
 
 PRODUCT_COPY_FILES += \
diff --git a/phh-on-data.sh b/phh-on-data.sh
index 472d16e..9c72782 100644
--- a/phh-on-data.sh
+++ b/phh-on-data.sh
@@ -3,25 +3,34 @@
 vndk="$(getprop persist.sys.vndk)"
 [ -z "$vndk" ] && vndk="$(getprop ro.vndk.version |grep -oE '^[0-9]+')"
 
-if getprop persist.sys.phh.no_vendor_overlay |grep -q true;then
+if getprop persist.sys.phh.no_vendor_overlay |grep -q true; then
 	for part in odm vendor;do
 		mount /mnt/phh/empty_dir/ /$part/overlay
 	done
 fi
 
-if getprop persist.sys.phh.caf.media_profile |grep -q true;then
+if getprop persist.sys.phh.caf.media_profile |grep -q true; then
     setprop media.settings.xml "/vendor/etc/media_profiles_vendor.xml"
 fi
 
+ 
+if getprop persist.sys.phh.no_stock_apps |grep -q true; then
+	for part in odm vendor; do
+		mount /mnt/phh/empty_dir/ /$part/overlay
+		mount /mnt/phh/empty_dir/ /$part/app
+	done
+fi
+
+
 
 minijailSrc=/system/system_ext/apex/com.android.vndk.v28/lib/libminijail.so
 minijailSrc64=/system/system_ext/apex/com.android.vndk.v28/lib64/libminijail.so
-if [ "$vndk" = 27 ];then
+if [ "$vndk" = 27 ]; then
     mount $minijailSrc64 /vendor/lib64/libminijail_vendor.so
     mount $minijailSrc /vendor/lib/libminijail_vendor.so
 fi
 
-if [ "$vndk" = 28 ];then
+if [ "$vndk" = 28 ]; then
     mount $minijailSrc64 /vendor/lib64/libminijail_vendor.so
     mount $minijailSrc /vendor/lib/libminijail_vendor.so
     mount $minijailSrc64 /system/lib64/vndk-28/libminijail.so
diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 4371632..ac38415 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -195,6 +195,86 @@ if [ "$1" == "persist.sys.phh.backlight.scale" ];then
     exit
 fi
 
+ 
+if [ "$1" == "persist.sys.phh.qin.dt2w" ];then
+    if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
+        exit 1
+    fi
+
+    echo "$prop_value" > /sys/devices/platform/soc/soc:ap-apb/70800000.i2c/i2c-3/3-0038/fts_gesture_mode
+    exit
+fi
+
+#safetynet/securize
+if [ "$1" == "persist.sys.phh.securize" ]; then
+    if [[ "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/secure ]; then
+        mkdir -p /data/adb/phh
+        cp /system/phh/secure.sh /data/adb/phh/secure
+    fi
+    /system/bin/sh /data/adb/phh/secure
+    exit
+fi
+
+#autorun
+if [ "$1" == "persist.sys.phh.autorun" ]; then
+    if [[ "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/run ]; then
+        mkdir -p /data/adb/phh
+        touch /data/adb/phh/run
+    fi
+    /system/bin/sh /data/adb/phh/run
+    exit
+fi
+
+#nolog
+if [ "$1" == "persist.sys.phh.nolog" ]; then
+    if [[ "$prop_value" != "false" && "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        setprop ctl.stop logd
+        setprop ctl.stop traced
+        setprop ctl.stop traced_probes
+    else
+        setprop ctl.start traced_probes
+        setprop ctl.start traced
+        setprop ctl.start logd
+    fi
+    exit
+fi
+
+#restart_sysui
+if [ "$1" == "sys.phh.restart_sysui" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        pkill systemui
+    fi
+    exit
+fi
+
+#dump_logs
+if [ "$1" == "sys.phh.dump_logs" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        logcat -b all -d > /sdcard/logs.txt
+    fi
+    exit
+fi
+
 if [ "$1" == "persist.sys.phh.disable_soundvolume_effect" ];then
     if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
         exit 1
diff --git a/rw-system.sh b/rw-system.sh
index 8b1d9e8..3cafeed 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -60,7 +60,17 @@ if [ "$vndk" = 26 ];then
 fi
 
 setprop sys.usb.ffs.aio_compat true
+setprop persist.adb.nonblocking_ffs false
 
+# Redmi Note 9S/Pro/Max props
+if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/curtana -e redmi/joyeuse -e redmi/excalibur; then
+    setprop ro.surface_flinger.has_HDR_display true
+    setprop ro.surface_flinger.has_wide_color_display true
+    setprop persist.device_config.runtime_native.usap_pool_enabled true
+    setprop debug.sf.latch_unsignaled 1
+    setprop debug.sf.enable_gl_backpressure 1
+fi
+    
 if getprop ro.vendor.build.fingerprint | grep -q -i -e Blackview/BV9500Plus;then
     setprop persist.adb.nonblocking_ffs true
 else
@@ -436,6 +446,21 @@ if getprop ro.vendor.build.fingerprint | grep -q -i \
     setprop persist.sys.qcom-brightness "$(cat /sys/class/leds/lcd-backlight/max_brightness)"
 fi
 
+#Xiaomi Redmi Note 9, Redmi 9 (Prime)
+if getprop ro.vendor.build.fingerprint |grep -qi -e Redmi/merlin -e Redmi/lancelot -e Redmi/angelican -e Redmi/dandelion; then
+    setprop persist.sys.overlay.devinputjack true
+fi
+
+# UMIDIGI A3X & A9
+if getprop ro.vendor.build.fingerprint |grep -qi -e UMIDIGI/A3X -e UMIDIGI/A9; then
+    setprop persist.sys.overlay.devinputjack true
+fi
+
+# CUBOT KINGKONG MINI2
+if getprop ro.vendor.build.fingerprint |grep -qi -e CUBOT/KINGKONG_MINI2; then
+    setprop persist.sys.overlay.devinputjack true
+fi
+
 #Realme 6
 if getprop ro.vendor.product.device |grep -iq -e RMX2001 -e RMX2151 -e RMX2111 -e RMX2111L1;then
     setprop persist.sys.phh.fingerprint.nocleanup true
@@ -629,6 +654,7 @@ if getprop ro.vendor.build.fingerprint | grep -qE -e ".*(crown|star)[q2]*lte.*"
     done
 fi
 
+
 # This matches both Razer Phone 1 & 2
 if getprop ro.vendor.build.fingerprint |grep -qE razer/cheryl;then
 	setprop ro.audio.monitorRotation true
@@ -657,6 +683,10 @@ if [ $(find /vendor/etc/audio -type f |wc -l) -le 3 ];then
 	mount -o bind /mnt/phh/empty_dir /vendor/etc/audio || true
 fi
 
+if ! grep android.hardware.ir /vendor/manifest.xml;then
+    mount -o bind system/phh/empty /system/etc/permissions/android.hardware.consumerir.xml
+fi
+
 for f in /vendor/lib{,64}/hw/com.qti.chi.override.so /vendor/lib{,64}/libVD*;do
     [ ! -f $f ] && continue
     # shellcheck disable=SC2010
@@ -714,6 +744,10 @@ if getprop ro.vendor.build.fingerprint | grep -qiE '^samsung/' && [ "$vndk" -ge
     fi
 fi
 
+if getprop ro.vendor.build.fingerprint | grep -q -e Sony/aosp_f5121/suzu; then
+    chown system:system /sys/class/leds/*/breath
+fi
+
 # For Nubia Red Magic 6 audio policy configuration
 if getprop ro.vendor.build.fingerprint | grep -q -e nubia/NX669; then
     umount /vendor/etc/audio
@@ -736,7 +770,8 @@ copyprop() {
         resetprop_phh "$1" "$(getprop "$2")"
     fi
 }
-if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ];then
+
+if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ] || [ -f /data/adb/phh/secure ];then
     copyprop ro.build.device ro.vendor.build.device
     copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
     copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
@@ -762,6 +797,7 @@ if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ];then
     copyprop ro.product.manufacturer ro.vendor.product.manufacturer
     copyprop ro.system.product.manufacturer ro.product.vendor.manufacturer
     copyprop ro.product.manufacturer ro.product.vendor.manufacturer
+
     (getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
         [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
     done
@@ -797,6 +833,17 @@ else
     mount -o bind /mnt/phh/xbin /system/xbin
 fi
 
+# Fix the google hotword detection for android 11 on redmi note 9s
+if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/curtana; then
+     resetprop_phh ro.product.model "Note 9S"
+fi
+if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/joyeuse; then
+     resetprop_phh ro.product.model "Note 9S Pro"
+fi
+if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/excalibur; then
+     resetprop_phh ro.product.model "Note 9S Pro Max"
+fi
+
 for abi in "" 64;do
     f=/vendor/lib$abi/libstagefright_foundation.so
     if [ -f "$f" ];then
@@ -989,14 +1036,26 @@ resetprop_phh ro.bluetooth.library_name libbluetooth.so
 
 board="$(getprop ro.board.platform)"
 
+if getprop ro.vendor.build.fingerprint |grep -iq xiaomi/cepheus;then
+	setprop ro.netflix.bsp_rev Q855-16947-1
+fi
+    
 if [ "$board" = atoll ] || [ "$board" = sm6250 ]; then
 	setprop ro.netflix.bsp_rev Q6250-19132-1
 fi
 
+if getprop ro.vendor.build.fingerprint |grep -qi redmi/curtana -e redmi/joyeuse -e redmi/excalibur; then
+	setprop ro.netflix.bsp_rev Q6250-19132-1
+fi
+
 if [ "$board" = msmnile ]; then
 	setprop ro.netflix.bsp_rev Q855-16947-1
 fi
 
+if getprop ro.vendor.build.fingerprint |grep -iq xiaomi/renoir;then
+	setprop ro.netflix.bsp_rev Q875-32774-1
+fi
+
 if [ "$board" = sm6150 ]; then
 	setprop ro.netflix.bsp_rev Q6150-17263-1
 fi
diff --git a/secure.sh b/secure.sh
new file mode 100644
index 0000000..98027e7
--- /dev/null
+++ b/secure.sh
@@ -0,0 +1,16 @@
+(getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
+    [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
+done
+resetprop_phh ro.build.tags release-keys
+resetprop_phh ro.boot.vbmeta.device_state locked
+resetprop_phh ro.boot.verifiedbootstate green
+resetprop_phh ro.boot.flash.locked 1
+resetprop_phh ro.boot.veritymode enforcing
+resetprop_phh ro.boot.warranty_bit 0
+resetprop_phh ro.warranty_bit 0
+resetprop_phh ro.debuggable 0
+resetprop_phh ro.secure 1
+resetprop_phh ro.build.type user
+resetprop_phh ro.build.selinux 0
+resetprop_phh ro.adb.secure 1
+setprop ctl.restart adbd
diff --git a/vndk.rc b/vndk.rc
index d1fffde..aa6e247 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -43,6 +43,9 @@ on property:persist.sys.phh.caf.audio_policy=*
 on property:persist.sys.phh.vsmart.dt2w=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.vsmart.dt2w"
 
+on property:persist.sys.phh.qin.dt2w=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.qin.dt2w"
+
 on property:persist.sys.phh.backlight.scale=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.backlight.scale"
 
@@ -61,6 +64,22 @@ on property:persist.sys.phh.disable_a2dp_offload=1
 on property:persist.sys.phh.disable_a2dp_offload=*
     setprop persist.bluetooth.bluetooth_audio_hal.disabled ${persist.sys.phh.disable_a2dp_offload}
 
+on property:persist.sys.phh.securize=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.securize"
+
+on property:persist.sys.phh.autorun=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.autorun"
+
+on property:persist.sys.phh.nolog=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.nolog"
+
+on property:sys.phh.restart_sysui=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.restart_sysui"
+
+on property:sys.phh.dump_logs=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.dump_logs"
+
+
 on property:init.svc.ril-proxy=stopped && property:persist.sys.phh.restart_ril=true
     start ril-proxy
 
-- 
2.25.1

